src/components/Button/index.tsx

```diff
@@ -342,18 +304,18 @@ function Button({
     );
 }
 
-Button.propTypes = propTypes;
-Button.defaultProps = defaultProps;
 Button.displayName = 'Button';
 
-const ButtonWithRef = React.forwardRef((props, ref) => (
-    <Button
-        // eslint-disable-next-line react/jsx-props-no-spreading
-        {...props}
-        forwardedRef={ref}
-    />
-));
+function ButtonWithRef(props: ButtonProps, ref: ForwardedRef<View>) {
+    return (
+        <Button
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
+            forwardedRef={ref}
+        />
+    );
+}
 
 ButtonWithRef.displayName = 'ButtonWithRef';
 
-export default withNavigationFallback(ButtonWithRef);
+export default withNavigationFallback(React.forwardRef(ButtonWithRef));
```

Comment:
If you add ref as a second param you can do it this way
```suggestion
export default withNavigationFallback(React.forwardRef(Button));
```